servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/GrpcSpanStatusExtractor.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2025 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicetalk.opentelemetry.http;
+
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanStatusBuilder;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanStatusExtractor;
+
+import javax.annotation.Nullable;
+
+/**
+ * Span status extractor for gRPC requests using gRPC status codes.
+ * <p>
+ * This extractor follows gRPC semantic conventions for determining span status
+ * based on gRPC status codes rather than HTTP status codes.
+ */
+final class GrpcSpanStatusExtractor implements SpanStatusExtractor<RequestInfo, GrpcTelemetryStatus> {
+
+    static final GrpcSpanStatusExtractor INSTANCE = new GrpcSpanStatusExtractor();
+
+    private GrpcSpanStatusExtractor() {
+    }
+
+    @Override
+    public void extract(
+            SpanStatusBuilder spanStatusBuilder,
+            RequestInfo requestInfo,
+            @Nullable GrpcTelemetryStatus response,
+            @Nullable Throwable error) {
+        if (error != null) {
```

Comment:
Another case for an error is a non-200 HTTP response status. grpc spec itself allows 415 response code + buggy proxies may return 5xx when they handle grpc 🫠